src/features/navigate.rs

```diff
@@ -28,6 +32,68 @@ pub fn make_navigate_regexp(config: &Config) -> String {
     )
 }
 
+// Create a less history file to be used by delta's child less process. This file is initialized
+// with the contents of user's real less hist file, to which the navigate regexp is appended. This
+// has the effect that 'n' or 'N' in delta's less process will search for the navigate regexp,
+// without the undesirable aspects of using --pattern, yet without polluting the user's less search
+// history with delta's navigate regexp. See
+// https://github.com/dandavison/delta/issues/237#issuecomment-780654036. Note that with the
+// current implementation, no writes to the delta less history file are propagated back to the real
+// history file so, for example, a (non-navigate) search performed in the delta less process will
+// not be stored in history.
+pub fn copy_less_hist_file_and_append_navigate_regexp(config: &Config) -> std::io::Result<PathBuf> {
+    let home_dir = dirs_next::home_dir().ok_or_else(|| Error::new(ErrorKind::NotFound, "Can't determine home dir"))?;
+    let initial_contents = "\
+.less-history-file:
+.search
+".to_string();
+    let contents = if let Some(hist_file) = get_less_hist_file(&home_dir) {
+        std::fs::read_to_string(hist_file).unwrap_or(initial_contents)
+    } else {
+        initial_contents
+    };
+    let mut hist_file = home_dir;
+    hist_file.push(".delta.lesshst");
+    writeln!(
+        std::fs::File::create(&hist_file)?,
+        "{}\"{}",
```

Comment:
@cben at the moment this just does the naive append. Can you show how to break this with `--set-marks` or similar?

Reply:
```
$ export LESS='--save-marks --status-column'
$ less README.md 
```
Inside less, press <kbd>d</kbd><kbd>M</kbd><kbd>a</kbd> to set a mark called `a` at end of screen.
Quit.
```
$ cat ~/.lesshst
.less-history-file:
.search
"foo
"bar
...
"various search strings...
.mark
m a 73 3692 /home/cben/term/delta/README.md
```
We can confirm the saved mark works:
```
$ less README.md 
```
Inside less press <kbd>'</kbd><kbd>a</kbd> to jump to mark `a`.  (and can see it highlighted in status column on left.)

Now use `delta --navigate`, e.g. by `git show`.  Navigation doesn't work.

Reply:
^ that was tested with the version when you merged this PR.  With current master it works :heavy_check_mark:.